src/main/java/org/rumbledb/expressions/control/TypeSwitchCaseExpression.java

```diff
@@ -5,20 +5,20 @@
 import org.rumbledb.expressions.Expression;
 import org.rumbledb.expressions.Node;
 import org.rumbledb.expressions.flowr.FlworVarSequenceType;
-import org.rumbledb.expressions.primary.VariableReference;
+import org.rumbledb.expressions.primary.VariableReferenceExpression;
 import sparksoniq.semantics.visitor.AbstractNodeVisitor;
 
 import java.util.ArrayList;
 import java.util.List;
 
 public class TypeSwitchCaseExpression extends Expression {
 
-    private VariableReference variableReferenceNode;
+    private VariableReferenceExpression variableReferenceNode;
     private List<FlworVarSequenceType> union = new ArrayList<>();
     private final Expression returnExpression;
 
     public TypeSwitchCaseExpression(
-            VariableReference var,
+            VariableReferenceExpression var,
```

Comment:
cosmetic: shall we use a fullname here instead of abbreviation, for compliance with rest of the project?